x-pack/plugin/runtime-fields/src/main/java/org/elasticsearch/xpack/runtimefields/mapper/IpFieldScript.java

```diff
@@ -111,4 +114,24 @@ public void emit(String v) {
             script.emit(v);
         }
     }
+
+    public static class EmitValues {
+        private final IpFieldScript script;
+
+        public EmitValues(IpFieldScript script) {
+            this.script = script;
+        }
+
+        public void emitFromPath(String path) {
+            for (Object v : script.extractFromSource(path)) {
+                if (v instanceof String) {
+                    try {
+                        script.emit(v.toString());
```

Comment:
why not casting to string at this point?

Reply:
May as well cast if you've already instaceofed